slobby/shop-be
# Task9

Task link: https://github.com/EPAM-JS-Competency-center/cloud-development-course-initial/blob/main/task9-bff-elastic-beanstalk/task.md#task-9-backend-for-frontend

Due time - 06.11.2021. Dead line - 06.11.2021 23:59

What were done:

1 - Create a bff-service.

2 - Product-service service API endpoint URL:  https://k9tqg1fapd.execute-api.eu-west-1.amazonaws.com/dev/products

- To get product use postman. 
method: GET
set url: https://k9tqg1fapd.execute-api.eu-west-1.amazonaws.com/dev/products

- To create product use postman. 
method: POST
set url: https://k9tqg1fapd.execute-api.eu-west-1.amazonaws.com/dev/products
no extra headers: only standart 7 headers.
Body: 
{
        "title": "BFF test",
        "description": "BFF test",
        "price": 555,
        "count": 555,
        "image": "https://s3uploader-s3uploadbucket-4dju787ezzkp.s3.eu-west-1.amazonaws.com/7.jpeg"
    }
All fields are required. If you want to load valid images please, chose any of these:
![image](https://user-images.githubusercontent.com/8141784/140622990-cd50b4a1-7e81-4ae0-91fe-9e2690678bb5.png)

3 - CART service API endpoint URL:  http://slobby-cart-api-dev.eu-west-1.elasticbeanstalk.com

4 - bff-service service URL:  http://slobby-bff-service-dev.eu-west-1.elasticbeanstalk.com

5 -  to call product-service and CART services via bff-service service URL.
   - http://slobby-bff-service-dev.eu-west-1.elasticbeanstalk.com/products
   - http://slobby-bff-service-dev.eu-west-1.elasticbeanstalk.com/cart

###Additional (optional) tasks:
#### 1. Add a cache at the bff-service.
 How to test:
- Open FE link: https://d1ddvc0h3kuyd3.cloudfront.net/
- in postman get products list
- in postman create new Product for instance:
{
        "title": "BFF test",
        "description": "BFF test",
        "price": 555,
        "count": 555,
        "image": "https://s3uploader-s3uploadbucket-4dju787ezzkp.s3.eu-west-1.amazonaws.com/4.jpeg"
    }
- Reopen FE link: https://d1ddvc0h3kuyd3.cloudfront.net/ and you see your new product
- in postman get products list - result shouldn’t have new product

Wait more than 2 minutes
- in postman get products list again - result should have new product


#### 2. Use NestJS to create bff-service instead of express

Self evaluation score - 7.

NB: 
ENV file exapmle:
CACHE_TTL=120
PORT=8080
NODE_ENV=development
products=your own
cart=your own
